preview/preview.js

```diff
@@ -1,24 +1,48 @@
-const d3 = require("d3");
-const SVGTypewriter = require("svg-typewriter");
+const { Typesetter } = require("@palantir/typesetter");
 
-function createUpdater(selector, options) {
+function createSvgUpdater(selector, options) {
     const element = document.querySelector(selector);
-    const selection = d3.select(element);
-    const writeOptions = Object.assign({}, {
-        selection: selection,
-        xAlign: "left",
-        yAlign: "top",
-        textRotation: 0
-    }, options);
+    const typesetter = Typesetter.svg(element);
+    const writeOptions = Object(options);
 
-    const measurer = new SVGTypewriter.CacheMeasurer(selection);
-    const wrapper = new SVGTypewriter.Wrapper();
-    const writer = new SVGTypewriter.Writer(measurer, wrapper);
+    const update = function() {
+        const rect = writeOptions.rect == null ? element.getBoundingClientRect() : writeOptions.rect;
+        element.innerHTML = "";
+        typesetter.write(this.text, rect.width, rect.height, this.options);
+    };
+
+    return {
+        update,
+        text: "",
+        options: writeOptions,
+    };
+}
+
+function retinaFix(ctx) {
```

Comment:
Oh this is pretty standard. We'll have to do the same when we start canvas rendering plottable.